keanu-project/src/test/java/io/improbable/keanu/algorithms/mcmc/MetropolisHastingsTest.java

```diff
@@ -31,33 +35,47 @@
     public DeterministicRule rule = new DeterministicRule();
 
     @Test
-    public void samplesContinuousPrior() {
+    public void samplesContinuousPriorSingleVariableSelected() {
 
-        DoubleVertex A = new GaussianVertex(20.0, 1.0);
-        DoubleVertex B = new GaussianVertex(20.0, 1.0);
+        MCMCTestCase testCase = new SumGaussianTestCase();
 
-        A.setValue(20.0);
-        B.setValue(20.0);
+        BayesianNetwork bayesNet = testCase.getModel();
+        KeanuProbabilisticModel model = new KeanuProbabilisticModel(bayesNet);
 
-        DoubleVertex Cobserved = new GaussianVertex(A.plus(B), 1.0);
+        NetworkSamples posteriorSamples = MetropolisHastings.builder()
+            .proposalDistribution(new PriorProposalDistribution(model.getLatentVertices()))
+            .rejectionStrategy(new RollBackToCachedValuesOnRejection(model.getLatentVertices()))
+            .variableSelector(MHStepVariableSelector.SINGLE_VARIABLE_SELECTOR)
+            .build()
+            .getPosteriorSamples(
+                model,
+                model.getLatentVertices(),
+                5000
+            );
 
-        Cobserved.observe(46.0);
+        testCase.assertExpected(posteriorSamples);
+    }
 
-        BayesianNetwork bayesNet = new BayesianNetwork(Arrays.asList(A, B, Cobserved));
-        bayesNet.probeForNonZeroProbability(100);
-        KeanuProbabilisticModel model = new KeanuProbabilisticModel(bayesNet);
+    @Test
+    public void samplesContinuousPriorAllVariablesSelected() {
 
-        NetworkSamples posteriorSamples = Keanu.Sampling.MetropolisHastings.withDefaultConfigFor(model).getPosteriorSamples(
-            model,
-            Arrays.asList(A, B),
-            5000
-        );
+        MCMCTestCase testCase = new SumGaussianTestCase();
 
-        double averagePosteriorA = posteriorSamples.getDoubleTensorSamples(A).getAverages().scalar();
-        double averagePosteriorB = posteriorSamples.getDoubleTensorSamples(B).getAverages().scalar();
+        BayesianNetwork bayesNet = testCase.getModel();
+        KeanuProbabilisticModel model = new KeanuProbabilisticModel(bayesNet);
 
-        double actual = averagePosteriorA + averagePosteriorB;
-        assertEquals(44.0, actual, 0.1);
+        NetworkSamples posteriorSamples = MetropolisHastings.builder()
+            .proposalDistribution(new PriorProposalDistribution(model.getLatentVertices()))
+            .rejectionStrategy(new RollBackToCachedValuesOnRejection(model.getLatentVertices()))
```

Comment:
Would it be helpful for Python users to be able to set the rejection strategy?  They can't do that at the moment.